dxr/static/js/dxr.js

```diff
@@ -339,8 +350,31 @@ $(function() {
                 results[result].pathLine = resultHead[1];
             }
 
-            var container = append ? contentContainer : contentContainer.empty();
-            container.append(nunjucks.render(tmpl, data));
+            if (!append) {
+                contentContainer
+                    .empty()
+                    .append(nunjucks.render('partial/results_container.html', data));
+            } else {
+                var resultsList = contentContainer.find('.results');
+
+                // If the first result is already on the page (meaning we showed
+                // some of its lines but just fetched more), add new lines to
+                // it instead of adding a new result section.
+                var firstResult = data.results[0];
+                var domFirstResult = resultsList.find(
+                    '.result[data-path="' + firstResult.path + '"]');
```

Comment:
I'm going to make a few comments just for "here's something to think about" purposes, but they shouldn't block merging this.

Comment one: this is probably a linear search, depending on what kind of data structures the browser uses. We could speed it up, if we need to, by squirreling away the path of the last result in a known place somewhere (or even just looking it up `data-path` on the last result and comparing against that).